src/status.ts

```diff
@@ -0,0 +1,130 @@
+  
+      // set up intermittent saves to the db for each robot
+      setInterval(this.saveClientInformation.bind(this), StatusManager.CLIENT_SAVE_INTERVAL);
+      this.connected = true;
+    });
+  }
+
+  handleDbConnError() {
+    // idk, log stuff?
+    this.logger?.log("Error connecting to database");
+  }
+
+  handleNewConnection(sender: WebSocket, clientIp: string, logger?: Logger) {
+    logger?.logOnce(`StatusManager got connection at ip ${clientIp}`);
+    // Check if this client is one we know about already
+    if (this.clientInformation.has(clientIp)) {
+      logger?.logOnce(`New connection for client ${clientIp}, which is already being tracked.`);
+    } else {
+      logger?.logOnce(`New connection for client ${clientIp}, which is not being tracked.`);
+    }
+  }
+
+  async saveClientInformation() {
+    // this.logger?.log("Persisting state to db...");
+
+    let batch = robotDb().batch();
+    await Promise.all(Array.from(this.clientInformation).map(async ([ip, client], _) => {
+      try {
+        let robot = RobotInformation.fromJSON((await robotDb().get(ip)).toString()) //this doesn't come out as the correct
+        // Don't save if nothing has changed; in fact, let's delete this robot from our local tracking because it's offline now
+        if (client.lastUpdated.toString() == robot.lastUpdated.toString()) {
+          // this.logger?.log(`Clearing tracking for expired client (${client.ip}, ${client.name}).`);
+          // this.clientInformation.delete(client.ip);
+          return;
+        }
+        robot.lastStatus = client.lastStatus;
+        robot.lastLocation = client.lastLocation;
+        robot.lastUpdated = client.lastUpdated;
+        
+        batch = batch.put(ip, robot.jsonString());
+      } catch (err) {
+        if (err.notFound) {
+          batch = batch.put(ip, client.jsonString());
+        } else {
+          throw err;
+        }
+      }
+    }));
+    return await batch.write();
+  }
+
+  private getNameFromTopic(topic: string) {
+    // Absolute vs relative path
+    if (topic.startsWith('/')) {
+      return topic.split('/')[1];
+    } else {
+      return topic.split('/')[0];
+    }
+  }
+
+  /**
+   * Handle any Robofleet message and, if it has the type RobofleetStatus
+   * (on any topic), handles the status management for the given sender  .
+   * 
+   * @param buf ByteBuffer containing the message data 
+   * @param clientIp string ip addr of the client sending the message
+   * @param name string name of the client sending the message
+   */
+  handleMessageBuffer(sender: WebSocket, buf: flatbuffers.ByteBuffer, clientIp: string, topic: string, logger?: Logger) {
+    if (!this.connected) {
+      this.logger?.log(`Not connected to database`);
+      return;
+    }
+
+    const metadata = getRobofleetMetadata(buf);
+    if (metadata?.type() === "amrl_msgs/RobofleetStatus" || metadata?.type() === "RobofleetStatus" ) {
```

Comment:
technically the type `RobofleetStatus` (as opposed to `amrl_msgs/RobofleetStatus`) is not valid; we should open an issue to fix this if we use it anywhere.

Reply:
This was an unnecessary part of the the conditional, I'll drop the second part.